src/tests/functional/test_main.py

```diff
@@ -29,25 +31,234 @@
 
 LOG_FILE = os.path.join(os.path.dirname(os.path.realpath(__file__)), "log.txt")
 
-BIN = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../../bin/cbc-binary-analysis")
-
-
+# BIN = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../../bin/cbc-binary-analysis")
+#
+# SRC = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../../")
 # Clear log file for each run
 open(LOG_FILE, "w").close()
 
 
+@pytest.fixture()
+def auth_token(pytestconfig):
+    """Get API token from command line"""
+    return pytestconfig.getoption("token")
+
+
+def _format_config(auth_token, feed_id):
+    """Configure for most of the test cases in this module."""
+    return (f"""id: cbc_binary_toolkit
+version: 0.0.1
+carbonblackcloud:
+  url: https://defense-eap01.conferdeploy.net
+  api_token: {auth_token}
+  org_key: WNEXFKQ7
+  ssl_verify: True
+  expiration_seconds: 3600
+database:
+  _provider: cbc_binary_toolkit.state.builtin.Persistor
+  location: ":memory:"
+engine:
+  name: Yara
+  feed_id: {feed_id}
+  type: local
+  _provider: cbc_binary_toolkit_examples.engine.yara_local.yara_engine.YaraFactory
+  rules_file: __file__/example_rule.yara
+    """)
+
+
+def _format_invalid_config(auth_token, feed_id):
+    """Configure for invalid test cases in this module."""
+    return (f"""id: cbc_binary_toolkit
+version: 0.0.1
+carbonblackcloud:
+  url: https://defense-eap01.conferdeploy.net
+  api_token: {auth_token}
+  org_key: WNEXFKQ7
+  ssl_verify: True
+  expiration_seconds: 3600
+database:
+  _provider: cbc_binary_toolkit.state.builtin.Persistor
+  location: ":memory:"
+engine:
+  name:
+  feed_id: {feed_id}
+  type: local
+  _provider: cbc_binary_toolkit_examples.engine.yara_local.yara_engine.YaraFactory
+  rules_file: __file__/example_rule.yara
+    """)
+
+
+def _create_feed(auth_token):
+    """Create a Feed to use for testing"""
+    # Feed Creation
+    url = "https://defense-eap01.conferdeploy.net/threathunter/feedmgr/v2/orgs/WNEXFKQ7/feeds"
+
+    payload = ("{\"feedinfo\": {\"name\": \"BATFuncTest\", \"owner\": \"DevRel\","
```

Comment:
You should be able to create this as a dictionary which will make it easier to read and use the json property in the request function
```
>>> url = 'https://api.github.com/some/endpoint'
>>> payload = {'some': 'data'}

>>> r = requests.post(url, json=payload)
```